wagtail/admin/templates/wagtailadmin/shared/dialog/dialog_toggle.html

```diff
@@ -5,8 +5,9 @@
     dialog_id - The ID of the dialog you are toggling open
     class_name - CSS classes for styling the button
     text - The text of the button
+    attr - Custom attributes for the button
 
 {% endcomment %}
-<button type="button" class="{{ class_name }}" data-a11y-dialog-show="{{ dialog_id }}">
+<button type="button" class="{{ class_name }}" data-a11y-dialog-show="{{ dialog_id }}" {{ attr }}>
```

Comment:
Adding `attr` here so we can add add `data-` attributes to dialog toggles to make it easier to reuse existing modal triggers.

Reply:
I’ve proposed an alternative, let me know what you think – am also happy to go for this as-is if that makes more sense to you.